src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_posix.cc

```diff
@@ -497,5 +496,11 @@ void BluetoothDeviceScanner::BluetoothDeviceScannerDelegate::threadMain() {
   scanner_->Thread();
 }
 
+void BluetoothDeviceScanner::BluetoothDeviceScannerDelegate::exitThreadMain() {
+  scanner_->shutdown_requested_ = true;
```

Comment:
@pestOO According to the description of the used BlueZ functions - it is guaranteed.